fix the vibecoded ui

M1noa opened the pull request:
i redesigned it and made them call a central style.css file instead of using style tags in the files themselves

<img width="1178" height="553" alt="image" src="https://github.com/user-attachments/assets/89527dad-9c15-47ca-8e7b-e9f9587b54d8" />

i also added auto downloading and verifying for the hugging face models in install.sh & wsl-setup.bat

M1noa committed:
fix the shitty vibecoded ui

screenshot: https://your-girlfriend.strokes-to-my.pics/u3w0fcmm.png

M1noa committed:
forgot i changed the port... changing it back to :5000

M1noa committed:
Merge branch 'Mr-Homeless:main' into main

M1noa committed:
install.sh and wsl-setup.bat install the ai model

and wsl-install.bat auto restarts if not interupted after 15s timer starts

M1noa committed:
Merge branch 'main' of https://github.com/M1noa/waldo

StroepWafel commented:
Danny you beat me to it, I was gonna do this lol, saw it in the video and went 'eww thats gross'

cwdaniel commented:
This PR goes beyond its stated scope of “fixing the vibe-coded UI.” It also introduces:

- Autodownload functionality in the install scripts.
- Readme language updates.

These changes should ideally be separated into distinct PRs for clarity and reviewability.

### Concerns and Recommendations

**Model Download**

- The script auto-downloads a 1.9 GB model from Hugging Face without checksum or signature verification, and only checks file size.
- This is brittle and unsafe from a supply-chain perspective.
- Recommendation: At minimum, verify a SHA-256 hash, pin to a specific revision/commit, and handle failures cleanly.


**Repository Hygiene**

- The script writes d`eepcheat/vit_g_ps14_ak_ft_ckpt_7_clean.pth` locally, but `.gitignore `doesn’t exclude it.
-` .pyd ` appears duplicated as well.
- Recommendation: Add ignores for `deepcheat/**.pth` (or the directory) to prevent accidental commits, and remove duplicate ignore patterns.

**UI JavaScript Performance & Accessibility**

-` static/main.js` attaches `mousemove `listeners to every `.card`, `.panel`, `.interactive`, and `<tr>`, updating CSS variables on each frame.
- On pages with many rows/cards, this causes jank and degrades performance. It also harms keyboard-only users.
- Recommendation:
-- Throttle or wrap updates in `requestAnimationFrame`.
-- Scope listeners to only the necessary interactive elements (avoid blanket `<tr>` bindings).
-- Ensure keyboard/focus styles provide parity with hover effects.

M1noa committed:
remove main.js and update install/model setup scripts

Deleted static/main.js, modified install.sh and wsl-setup.bat to verify model file integrity via SHA-256 after download, updated .gitignore to exclude model files and temporary directories

fixed all things mentioned in https://github.com/Mr-Homeless/waldo/pull/585#issuecomment-3362478481

M1noa commented:
fixed it all in the latest commit [a6edcb548e5c6b437f60bf73d945df325648848c](https://github.com/Mr-Homeless/waldo/pull/585/commits/a6edcb548e5c6b437f60bf73d945df325648848c)

M1noa commented:
you could also automate the shai-256 part via using [the git lfs info](https://huggingface.co/jinggu/jing-model/raw/main/vit_g_ps14_ak_ft_ckpt_7_clean.pth)
